nextcord/cache.py

```diff
@@ -0,0 +1,141 @@
+# SPDX-License-Identifier: MIT
+
+from enum import Enum
+from time import time as now
+from typing import Hashable, Optional, Union
+
+
+class CacheMissing(Enum):
+    MISSING = 0
+
+
+_MISSING = CacheMissing.MISSING
+
+
+class SizedDict(dict):
+    """This represents a sized cache dictionary where objects over size limit
+    will be removed upon adding a new object.
+
+    .. note::
+        This cache relies on the presumed order of dictionaries after Python 3.7
+        This allows for the cache to be much faster and take 3x less memory than
+        the TTL cache from the `collections` module.
+    """
+
+    __slots__ = ("maxsize",)
+
+    def __init__(self, maxsize: Optional[Union[float, int]] = None) -> None:
+        self.maxsize: Union[float, int] = maxsize or float("inf")
+
+    def __setitem__(self, key, val) -> None:
+        """Ensures doesn't exceed max size before adding new item."""
+        if len(self) >= self.maxsize:
+            for item in self:  # iter to delete first item
+                del self[item]
+                break
+        super().__setitem__(key, val)

```

Comment:
I know you just changed it, but if speed is an issue would doing `dict.__setitem__` adequately speed things up?

Reply:
> I know you just changed it, but if speed is an issue would doing `dict.__setitem__` adequately speed things up?

Yes, `dict.__setitem__` is significantly faster than `super()`, only very slightly behind a relay variable due to attribute lookup, it would be a preferred approach if I could.